src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -706,8 +712,20 @@ def __call__(
                     noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
                     noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
 
+                    # Based on 3.4. in https://arxiv.org/pdf/2305.08891.pdf
```

Comment:
Maybe we can do a `if guidance_rescale != 0` check to avoid unnecessary computation.